runtime/port/unix/j9sysinfo.c

```diff
@@ -1619,3 +1719,95 @@ j9sysinfo_get_cache_info(struct J9PortLibrary *portLibrary, const J9CacheInfoQue
 	Trc_PRT_sysinfo_get_cache_info_exit(result);
 	return result;
 }
+
+/*
+ * Initializes the struct J9CgroupMetricIteratorState attributes basing on the subsystem
+ */
+
+int32_t
+j9sysinfo_cgroup_subsystem_iterator_init(struct J9PortLibrary *portLibrary, uint64_t subsystem, struct J9CgroupMetricIteratorState *state)
+{
+	Assert_PRT_true(NULL != state);
+	state->count = 0;
+	switch (subsystem) {
+		case OMR_CGROUP_SUBSYSTEM_MEMORY :
+			 state->numElements = sizeof(j9CgroupMemoryMetricMap) / sizeof(j9CgroupMemoryMetricMap[0]);
+			 state->subsystemid = OMR_CGROUP_SUBSYSTEM_MEMORY;
+			 break;
+		case OMR_CGROUP_SUBSYSTEM_CPU :
+			 state->numElements = sizeof(j9CgroupCpuMetricMap) / sizeof(j9CgroupCpuMetricMap[0]);
+			 state->subsystemid = OMR_CGROUP_SUBSYSTEM_CPU;
+			 break;
+		case OMR_CGROUP_SUBSYSTEM_CPUSET :
+			 state->numElements = sizeof(j9CgroupCpusetMetricMap) / sizeof(j9CgroupCpusetMetricMap[0]);
+			 state->subsystemid = OMR_CGROUP_SUBSYSTEM_CPUSET;
+			 break;
+		default :
+			return J9PORT_ERROR_SYSINFO_NOT_SUPPORTED;
+	}
+	return 0;
+}
+
+/*
+ * Checks if there is a next attribute subsystem
+ */
+
+BOOLEAN
+j9sysinfo_cgroup_subsystem_iterator_hasNext(struct J9PortLibrary *portLibrary, const struct J9CgroupMetricIteratorState *state)
+{
+	return (state->count < state->numElements);
+}
+
+/*
+ * Loads the next key and value to the struct J9CgroupMetricElement
+ */
+
+int32_t
+j9sysinfo_cgroup_subsystem_iterator_next(struct J9PortLibrary *portLibrary, struct J9CgroupMetricIteratorState *state, struct J9CgroupMetricElement *metricElement, BOOLEAN *printUnits)
+{
+	int32_t rc = J9PORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+	if (state->count > state->numElements) {
+		goto _end;
+	}
+	FILE *file = NULL;
+	OMRPORT_ACCESS_FROM_J9PORT(portLibrary);
+	J9CgroupSubsystemMetricMap *subsystemMetricMap = NULL;
+	switch (state->subsystemid) {
+		case OMR_CGROUP_SUBSYSTEM_MEMORY :
+			subsystemMetricMap = j9CgroupMemoryMetricMap;
+			break;
+		case OMR_CGROUP_SUBSYSTEM_CPU :
+			subsystemMetricMap = j9CgroupCpuMetricMap;
+			break;
+		case OMR_CGROUP_SUBSYSTEM_CPUSET :
+			subsystemMetricMap = j9CgroupCpusetMetricMap;
+			break;
+		default:
+			rc = J9PORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE;
+			goto _end;
+	}
+	metricElement->key = subsystemMetricMap[state->count].metricTag;
+	metricElement->units = subsystemMetricMap[state->count].metricUnit;
+	intptr_t fd = omrsysinfo_cgroup_get_handle_subsystem_file(state->subsystemid, subsystemMetricMap[state->count].metricFilename);
+	if (fd != 0) {
+		file = fdopen(fd, "r");
+		rc = readCgroupMetricFromFile(portLibrary, subsystemMetricMap[state->count].metricKeyInFile, file, metricElement->value);
+		if (subsystemMetricMap[state->count].isValueToBeChecked) {
+				int64_t result = 0;
+				sscanf(metricElement->value, "%"PRId64, &result);
```

Comment:
There must be a space between `"%"` and `PRId64`. The return value from `scanf` must be checked.